accessibility-best-practices.md

```diff
@@ -0,0 +1,243 @@
+
+Every form field (apart from buttons) must have a `label` that describes its purpose.
+
+Each `label` should be explicitly paired with its field by a `for` attribute on the `label` equal to the `id` on the field. Not only does this allow screen readers and other assistive technology to announce the correct label when the field receives focus, it also increases the size of the clickable area for other users.
+
+In the example below, the "Flying from?" label has as `for="location"` attribute which explicitly pairs it with the `select` control that has a corresponding `id="location"` attribute.
+
+```html
+<label for="location">Flying from?</label>
+<select name="location" id="location">
+    <option value="ABZ">Aberdeen</option>
+    <option value="BFS">Belfast International</option>
+    ...
+</select>
+```
+
+Placeholder text (a default vaule in the field) should **not** be used instead of a separate `label` element. Placeholders are cleared on focus and often not announced at all by screen readers or other assistive technology.
+
+### Hidden labels
+
+In some cases, it might be appropriate to visually hide a label. For example, where separate fields are provided for the day, month and year of a date, it is often unnecessary to display the label for each. In this situation, it is acceptable to hide the labels with CSS as long as they remain accessible to assistive technology in the HTML. See the section on [Hiding content in an accessible way](#hidden) for an example of this.
+
+```html
+<fieldset>
+    <legend>Date of birth</legend>
+    <label for="day" class="hidden">Day</label>
+    <input id="day" type="text">
+    <label for="month" class="hidden">Month</label>
+    <select id="month">
+        <option value="jan">January</option>
+        <option value="feb">February</option>
+        ...
+    </select>
+    <label for="year" class="hidden">Year</label>
+    <input id="year" type="text">
+</fieldset>
+```
+
+### Labels for checkboxes and radio buttons
+As well as being explicitly associated with `for` and `id` attributes, labels for checkboxes and radio buttons should appear *after* the control in the markup and be displayed *to the right* of the control on screen. This convention ensures compatibility with the widest range of assistive technologies, and is also consisitent with UI patterns established by Microsoft and Apple.
+
+### Identifying mandatory fields
+A mandatory field should be identified in its label, in a way that doesn't rely on vision, so that it can be announced by assistive technology when the field receives focus. This could be an asterisk (*) or simply say "required". Where an asterisk or other symbol is used, this should be explained at the top of the form (e.g. "fields marked with * are required.") Where all fields in a form are mandatory, it is acceptable to simply say "all fields are mandatory", but probably easier (technically) to adopt the same approach throughout and mark all individual fields.
```

Comment:
Mentioning the use of `<abbr title='required field'>*</abbr>` in required labels might be a nice touch here.